lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -26,37 +26,49 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel. Implicitly attaches the
-	 * Channel. However, if the channel is in or moves to the FAILED
-	 * state before the operation succeeds, it will result in an error
-	 * @return: the current present members.
+	 * String parameter names to get() call with Param[] as an argument
+	 */
+	public final static String GET_WAITFORSYNC = "waitForSync";
+	public final static String GET_CLIENTID = "clientId";
+	public final static String GET_CONNECTIONID = "connectionId";
+
+	/**
+	 * Get the presence state for this channel. Take Param[] array as an argument
+	 * @param params
+	 * @return
 	 * @throws AblyException
+	 * @throws InterruptedException
 	 */
-	public synchronized PresenceMessage[] get() throws AblyException {
+	public synchronized PresenceMessage[] get(Param... params) throws InterruptedException, AblyException {
```

Comment:
OK, so I missed the fact that the existing methods throw `InterruptedException`. But we're not consistent here, because we don't have those on other blocking methods such as the REST HTTP API (and the mainstream java blocking IO model doesn't routinely have `InterruptedException` on APIs that block for IO). You can argue that in the case of `presence.get()` we're not just blocking for IO, but we're performing a wait for a sequence of operations to complete, which includes IO; but i you look at higher level libraries (eg the Apache HttpClient) then you still don't have an `InterruptedException` on the blocking `HttpExecutor.execute()`; you just have `IOException`, when something to do with the IO failed, and `HttpException`, when the IO happened but there was a protocol error with the resulting data. We're wrapping both of those kinds of problems in an `AblyException`, for better or worse, and I don't think this `presence.get(waitForSync=true)` is in a different category.

As for the point that we're already declaring that as thrown, so existing callers that catch it will have to modify their code if we decide to remove them; I think it's ok. It is easier to remove a catch block from existing code than it is to add one and think about what to do with the exception.

Also, I'm not sure that `InterruptedException` is even semantically the correct thing if a thread blocked on the `presence.get()` is interrupted - I would say that `InterruptedIOException` is probably more correct (and this, being an `IOException`, is already wrapped as `AblyException`)